### ISSUE TITLE
For orgDefault, quotas r adding for same api instead of replacing the existing one

### ISSUE THREAD

**Issue by hakila** [Axway-API-Management-Plus/apimanager-swagger-promote#216]
**Swagger-Promote version**
> Please make sure to use the latest Swagger-Promote version.

**API-Manager and Service-Pack Version**
7.5.3

**Expected behavior**
quota should be updated instead of adding

**Actual behavior**
quota keeps adding for api when making chnages

**Log-Output in Debug**

**Test case sample**
> Please describe the steps you have done to replicate the issue
 I created api with default quota 3 per hour. Then I made chnage and put quota 100 per second. Instead of replacing 3per hour to 100 per second, script added another entry 100 per second. In this case which gets applied? This will be confusing to find out which will be applied. If one quota record is maintained per entry, that will help us. If that is the behaviour, can we have option to in script to get how we wanted..
![default quota mul entries for api](https://user-images.githubusercontent.com/17910073/70085853-11dfee80-15df-11ea-84ba-00325f05caec.png)

**Comment by cwiechmann** [comment]
I think this use-case is covered as part of the documentation here:
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/wiki/2.3-Configure-API-Quotas

> When using quotaMode=add (which is the default) and changing Quota-Keys (e.g. Type, Time-range, Per) in the API-Configuration file Swagger-Promote cannot identify anymore if existing quotas has been added manually or by Swagger-Promote before. Therefore new Quota-Entries will be added.
The recommendation is to avoid too many Key changes in the API-Config-File, but if it's required two options: 
> - replicate the API (maybe only once) with quotaMode=replace to bring it to the configured desired state
> - clean-up unused quotas in API-Manager manually

You have changed one of the keys (Time-Range) in your case which is used by Swagger-Promote to identify the existing entry. See the possible options. Perhaps that's helpful.

**Comment by hakila** [comment]
I tested just chnaged the quota with quotaMode=replace, it worked. Then i commented that quota mode and chnaged the quota. But still it behaves as replace. I see sometimes it is adding multiple. But this time, just chnaging quota for both add and replace, works as replace. Could you please test this
0      INFO  nvironmentProperties| Loaded environment properties from file: env.properties.
710    INFO     APIManagerAdapter| API-Manager version is: 7.5.3 SP6
774    INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: '/opt/axway/pythonscripts/conf/apiswagger/apicreatejson/TESTMinimumStayCategory06RulesCICD_transformed.json' (relative path)
775    INFO  IImportConfigAdapter| Handling configured client-applications.
823    INFO  IImportConfigAdapter| Found existing application: 'ATP_ATP_App' (null) based on given name 'ATP_ATP_App'
823    INFO  IImportConfigAdapter| Found existing application: 'X91_91_App' (null) based on given name 'X91_91_App'
823    INFO  IImportConfigAdapter| Found existing application: 'XAB_AB_App' (null) based on given name 'XAB_AB_App'
930    INFO     APIManagerAdapter| Found existing API on path: '/services/test/v1/rules/record3/cat6cicd' (published) (ID: '528c8de9-7b76-4d79-b232-4f81d5951eb9')
1770   INFO        APIChangeState| Changed property: applicationQuota[Desired: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=null, method=*, type=throttle, config={period=hour, per=1, messages=34}]]]' vs Actual: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=528c8de9-7b76-4d79-b232-4f81d5951eb9, method=*, type=throttle, config={messages=3, period=hour, per=1}]]]']
1771   INFO     APIManagerAdapter| Strategy: Going to update existing API: TEST Minimum Stay (Category 06) Rules CICD (Version: null)
1771   INFO     APIManagerAdapter| Recognized the following changes. Potentially Breaking: [] plus Non-Breaking: [applicationQuota]
1771   INFO     APIManagerAdapter| Strategy: Update existing API, as all changes can be applied in current state.
1808   INFO     APIManagerAdapter| Updating Application-Default-Quota for API: TEST Minimum Stay (Category 06) Rules CICD
1862   INFO     APIManagerAdapter| Application-Default quota successfully updated: TEST Minimum Stay (Category 06) Rules CICD
1862   INFO     APIManagerAdapter| All desired organizations: [ATP_ATP_Org, X91_91_Org, XAB_AB_Org, API Development] have already access. Nothing to do.
1863   INFO     APIManagerAdapter| All desired applications: [[ATP_ATP_App], [X91_91_App], [XAB_AB_App]] have already a subscription. Nothing to do.
1866   INFO                   App| Created API-Properties file: '/opt/axway/pythonscripts/conf/apiswagger/apiconfigjson/TESTMinimumStayCategory06RulesCICD_Details.prop'
1866   INFO                   App| Successfully replicated API-State into API-Manager
 Also we had issues with images, we r having white background and swagger process is manipulatting the image . There is case already opened. Becos of that even though there is no chnage, it always comes with image chnage. But in the above case i don't see the image chnage. Please investigate this also why image chnage is not coming. I will provide image bug number

**Comment by hakila** [comment]
Image bug number  is 112.

**Comment by cwiechmann** [comment]
Hi @hakila, Based on what I can in the log the behavior looks correct. 
This is what you have configured for your quota as you run Swagger-Promote:
**Desired:** config={period=hour, per=1, messages=34}
And this is the quota found for that API in API-Manager:
**Actual:** config={messages=3, period=hour, per=1}
What happens depends on the quotaMode. With set to __replace__:
You will end up with having only the **Desired** Quota: 34 Msg/1 hour the existing quota is removed.
With __add__ you will see two quota restrictions as the desired quota has been merged into the actual quota. 

If you think this doesn't work please provide more information. 
- Please let me know the Swagger-Promote version
- Turn on Debug for Swagger-Promote
- Execute and describe all steps (including some screenshots of the actual state)
- for each Swagger-Promote run, please share the Log-File

Concerning #112 - Again, Swagger-Promote is not changing the image, hence there is nothing Swagger-Promote can do. That's the reason why #112 is closed.
The API-Manager is converting the image when importing, if you would like to have this changed, please create an Axway support-case, however I'm not very confident that this change will ever be implemented.
